### ISSUE TITLE
Deliver fails to delete screenshots that don't yet exist when overwrite_screenshots is true

### ISSUE THREAD

**Issue by dlakes-hx** [fastlane/fastlane#17380]
Noticing an issue with how `deliver` is sending along screenshots. Looking through issues saw lots of activity around screenshots recently such as #17346, #16972, and #16842, but nothing that directly correlated.

### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

* Create a new version release in the app store connect
* add screenshots to `fastlane/screenshots` directory
* Run `deliver` with `overwrite_screenshots` set to true
* Receive error message: `[!] Failed verification of all screenshots deleted... ### screenshot(s) still exist`. Note the number indicated
* go to app store connect and confirm that no screenshots are present for the release
* go to `fastlane/screenshots` directory
* confirm that number of screenshots present equals count noted in prior step

##### Command executed

```ruby
deliver(
  ipa: "[REDACTED]",
  primary_category: "[REDACTED]",
  secondary_category: "[REDACTED]",
  overwrite_screenshots: true,
  force: true
)
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> 
[14:16:40]: ---------------------
[14:16:40]: --- Step: deliver ---
[14:16:40]: ---------------------
[14:16:40]: Login to App Store Connect [REDACTED]
[14:16:41]: Login successful

+--------------------------------------+----------------------------------------------------------------+
|                                        deliver 2.162.0 Summary                                        |
+--------------------------------------+----------------------------------------------------------------+
| ipa                                  | [REDACTED]                                                     |
| primary_category                     | [REDACTED]                                                     |
| secondary_category                   | [REDACTED]                                                     |
| overwrite_screenshots                | true                                                           |
| force                                | true                                                           |
| app_identifier                       | [REDACTED]                                                     |
| screenshots_path                     | ./fastlane/screenshots                                         |
| metadata_path                        | ./fastlane/metadata                                            |
| app_version                          | 4.0.1                                                          |
| username                             | [REDACTED]                                                     |
| platform                             | ios                                                            |
| edit_live                            | false                                                          |
| use_live_version                     | false                                                          |
| skip_binary_upload                   | false                                                          |
| skip_screenshots                     | false                                                          |
| skip_metadata                        | false                                                          |
| skip_app_version_update              | false                                                          |
| submit_for_review                    | false                                                          |
| reject_if_possible                   | false                                                          |
| automatic_release                    | false                                                          |
| phased_release                       | false                                                          |
| reset_ratings                        | false                                                          |
| run_precheck_before_submit           | true                                                           |
| precheck_default_rule_level          | warn                                                           |
| ignore_language_directory_validation | false                                                          |
| precheck_include_in_app_purchases    | true                                                           |
+--------------------------------------+----------------------------------------------------------------+

[14:16:43]: Making sure the latest version on App Store Connect matches '4.0.1'...
[14:16:44]: '4.0.1' is the latest version on App Store Connect
[14:16:44]: Loading './fastlane/metadata/en-US/description.txt'...
[14:16:44]: Loading './fastlane/metadata/en-US/keywords.txt'...
[14:16:44]: Loading './fastlane/metadata/en-US/support_url.txt'...
[14:16:44]: Loading './fastlane/metadata/en-US/marketing_url.txt'...
[14:16:44]: Loading './fastlane/metadata/en-US/promotional_text.txt'...
[14:16:44]: Loading './fastlane/metadata/en-US/privacy_url.txt'...
[14:16:44]: Loading './fastlane/metadata/copyright.txt'...
[14:16:44]: Loading './fastlane/metadata/review_information/first_name.txt'...
[14:16:44]: Loading './fastlane/metadata/review_information/last_name.txt'...
[14:16:44]: Loading './fastlane/metadata/review_information/phone_number.txt'...
[14:16:46]: Will begin uploading metadata for '4.0.1' on App Store Connect
[14:16:46]: Uploading metadata to App Store Connect for version
[14:16:49]: Uploading app review information to App Store Connect
[14:16:51]: Will begin uploading snapshots for '4.0.1' on App Store Connect
[14:16:51]: Starting with the upload of screenshots...
[14:16:54]: Number of screenshots not deleted: 35
[14:16:54]: Failed to delete all screenshots... Tries remaining: 4
[14:16:55]: Number of screenshots not deleted: 35
[14:16:55]: Failed to delete all screenshots... Tries remaining: 3
[14:16:55]: Number of screenshots not deleted: 35
[14:16:55]: Failed to delete all screenshots... Tries remaining: 2
[14:16:56]: Number of screenshots not deleted: 35
[14:16:56]: Failed to delete all screenshots... Tries remaining: 1
[14:16:57]: Number of screenshots not deleted: 35
[14:16:57]: Reporting build status error for lane: deploy_prod_ios
[14:16:57]: Backtrace: ["/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'", 
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'", 
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:104:in `delete_screenshots'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:36:in `upload'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/runner.rb:146:in `upload_metadata'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/runner.rb:54:in `run'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:17:in `run'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'",
  "Fastfile:446:in `block in parsing_binding'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane.rb:33:in `call'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'",
  "Fastfile:722:in `block in parsing_binding'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane.rb:33:in `call'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:45:in `execute'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/bin/fastlane:23:in `<top (required)>'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<main>'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `eval'",
  "/Users/[REDACTED]/jenkins/workspace/[REDACTED]/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `<main>'"]
[14:16:57]: No error info available. Returning the exception as message
[14:16:57]: Failed verification of all screenshots deleted... 35 screenshot(s) still exist
 </pre>
</details>

### Environment

<details>
<pre>
<summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.7                                     |
| Ruby                        | 2.6.3                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | ~/.rvm/rubies/ruby-2.6.3/bin/fastlane       |
| Host                        | Mac OS X 10.15.7 (19H2)                     |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.6.3/lib                |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.0.1                                      |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# [REDACTED LANES]

lane :deploy_prod_ios do |options|
  # Redacted setup steps including build & screenshot generation

  deliver(
    ipa: "[REDACTED]",
    primary_category: "[REDACTED]",
    secondary_category: "[REDACTED]",
    overwrite_screenshots: true,
    force: true
  )

  # Other redacated steps
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.162.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                            | Version | Update-Status |
| ------------------------------------------------- | ------- | ------------- |
| fastlane-plugin-cryptex                           | 0.1.5   | ✅ Up-To-Date  |
| fastlane-plugin-android_change_package_identifier | 0.1.0   | ✅ Up-To-Date  |
| fastlane-plugin-android_change_string_app_name    | 0.1.1   | ✅ Up-To-Date  |
| fastlane-plugin-msbuild                           | 0.1.4   | ✅ Up-To-Date  |
| fastlane-plugin-souyuz                            | 0.9.1   | ✅ Up-To-Date  |
| fastlane-plugin-appicon                           | 0.15.0  | ✅ Up-To-Date  |
| fastlane-plugin-xml_editor                        | 0.2.0   | ✅ Up-To-Date  |
| fastlane-plugin-unzip                             | 1.0.1   | ✅ Up-To-Date  |
| fastlane-plugin-appcenter                         | 1.9.0   | ✅ Up-To-Date  |
| fastlane-plugin-automated_test_emulator_run       | 1.6.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                               | Version      |
| ------------------------------------------------- | ------------ |
| did_you_mean                                      | 1.3.0        |
| executable-hooks                                  | 1.6.0        |
| bundler-unload                                    | 1.0.2        |
| rubygems-bundler                                  | 1.4.5        |
| bundler                                           | 2.1.4        |
| rake                                              | 13.0.1       |
| CFPropertyList                                    | 3.0.2        |
| public_suffix                                     | 4.0.6        |
| addressable                                       | 2.7.0        |
| atomos                                            | 0.1.3        |
| aws-eventstream                                   | 1.1.0        |
| aws-partitions                                    | 1.379.0      |
| aws-sigv4                                         | 1.2.2        |
| jmespath                                          | 1.4.0        |
| aws-sdk-core                                      | 3.109.0      |
| aws-sdk-kms                                       | 1.39.0       |
| aws-sdk-s3                                        | 1.82.0       |
| babosa                                            | 1.0.3        |
| claide                                            | 1.0.3        |
| colored                                           | 1.2          |
| colored2                                          | 3.1.2        |
| highline                                          | 1.7.10       |
| commander                                         | 4.4.6        |
| commander-fastlane                                | 4.4.6        |
| declarative                                       | 0.0.20       |
| declarative-option                                | 0.1.0        |
| digest-crc                                        | 0.6.1        |
| unf_ext                                           | 0.0.7.7      |
| unf                                               | 0.1.4        |
| domain_name                                       | 0.5.20190701 |
| dotenv                                            | 2.7.6        |
| emoji_regex                                       | 3.0.0        |
| excon                                             | 0.76.0       |
| multipart-post                                    | 2.0.0        |
| faraday                                           | 1.0.1        |
| http-cookie                                       | 1.0.3        |
| faraday-cookie_jar                                | 0.0.7        |
| faraday_middleware                                | 1.0.0        |
| fastimage                                         | 2.2.0        |
| gh_inspector                                      | 1.1.3        |
| jwt                                               | 2.2.2        |
| memoist                                           | 0.16.2       |
| multi_json                                        | 1.15.0       |
| os                                                | 1.1.1        |
| signet                                            | 0.14.0       |
| googleauth                                        | 0.13.1       |
| httpclient                                        | 2.8.3        |
| mini_mime                                         | 1.0.2        |
| uber                                              | 0.1.0        |
| representable                                     | 3.0.4        |
| retriable                                         | 3.1.2        |
| google-api-client                                 | 0.38.0       |
| google-cloud-env                                  | 1.3.3        |
| google-cloud-errors                               | 1.0.1        |
| google-cloud-core                                 | 1.5.0        |
| google-cloud-storage                              | 1.29.0       |
| json                                              | 2.3.1        |
| mini_magick                                       | 4.10.1       |
| plist                                             | 3.3.0        |
| rubyzip                                           | 2.3.0        |
| security                                          | 0.1.3        |
| naturally                                         | 2.2.0        |
| simctl                                            | 1.6.8        |
| slack-notifier                                    | 2.3.2        |
| terminal-notifier                                 | 2.0.0        |
| unicode-display_width                             | 1.7.0        |
| terminal-table                                    | 1.8.0        |
| tty-screen                                        | 0.8.1        |
| tty-cursor                                        | 0.7.1        |
| tty-spinner                                       | 0.9.3        |
| word_wrap                                         | 1.0.0        |
| nanaimo                                           | 0.3.0        |
| xcodeproj                                         | 1.18.0       |
| rouge                                             | 2.0.7        |
| xcpretty                                          | 0.3.0        |
| xcpretty-travis-formatter                         | 1.0.0        |
| fastlane-plugin-android_change_package_identifier | 0.1.0        |
| mini_portile2                                     | 2.3.0        |
| nokogiri                                          | 1.8.5        |
| fastlane-plugin-android_change_string_app_name    | 0.1.1        |
| fastlane-plugin-appcenter                         | 1.9.0        |
| fastlane-plugin-appicon                           | 0.15.0       |
| fastlane-plugin-automated_test_emulator_run       | 1.6.0        |
| fastlane-plugin-cryptex                           | 0.1.5        |
| fastlane-plugin-msbuild                           | 0.1.4        |
| souyuz                                            | 0.9.1        |
| fastlane-plugin-souyuz                            | 0.9.1        |
| fastlane-plugin-unzip                             | 1.0.1        |
| fastlane-plugin-xml_editor                        | 0.2.0        |
| simple_oauth                                      | 0.3.1        |
</details>

*generated on:* **2020-10-05**
</pre></details>

**Comment by squallstar** [comment]
I am having the same issue:

```
\Mon Oct 05 2020 05:00:46 GMT-0700 (PDT) - info: [05:00:46]: Number of screenshots not deleted: 30
|Mon Oct 05 2020 05:00:46 GMT-0700 (PDT) - info: [05:00:46]: Failed to delete all screenshots... Tries remaining: 4
/Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Number of screenshots not deleted: 30
Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Failed to delete all screenshots... Tries remaining: 3
-Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Number of screenshots not deleted: 30
Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Failed to delete all screenshots... Tries remaining: 2
\Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Number of screenshots not deleted: 30
Mon Oct 05 2020 05:00:47 GMT-0700 (PDT) - info: [05:00:47]: Failed to delete all screenshots... Tries remaining: 1
|Mon Oct 05 2020 05:00:48 GMT-0700 (PDT) - info: [05:00:48]: Number of screenshots not deleted: 30

Failed verification of all screenshots deleted... 30 screenshot(s) still exist
```

**Comment by rogerluan** [comment]
Tagging the screenshot :ninja: here: @ainame 😇

**Comment by ainame** [comment]
Haha thank you for tagging me. This sounds strange though🤔 

@dlakes-hx 

> Create a new version release in the app store connect

Did you have some existing screenshots in the previous version? Or did you add/remove regions to support screenshots?

In any cases, manually cleaning up screenshots on App Store Connect should resolve the issue for the time being, I believe.

**Comment by dlakes-hx** [comment]
@ainame yes, the prior (reviewed and released) version did have screenshots attached. We're attempting to perform a minor version bump, and the new minor release has no assets in the submission yet. We're not able to delete the screenshots for the prior version, because it's already through the submission process.

**Comment by ainame** [comment]
@dlakes-hx 

>  the new release has no assets in the submission yet

Does this mean that the new release you are preparing with `deliver` hasn't succeeded?

I can't tell you if this is the case for you, but there is a tricky behaviour in `overwrite_screenshots` option. If your current local screenshots' directory misses a language/region that was previously uploaded for, `deliver` will skip the deletion for `overwrite_screenshots`. So please check what you have locally and on App Store Connect...?

https://github.com/fastlane/fastlane/blob/5ee9f2e1f1d602e2af129e94eec2313091448abf/deliver/lib/deliver/upload_screenshots.rb#L87-L88

This is the only case I come up with that `deliver` may not be able to delete all the screenshots.

**Comment by dlakes-hx** [comment]
Yes @ainame it is failing in the `deliver` step. The only assets on the prior release are for `en-US`, and also the same on the pending release in the `fastlane/screenshots/en-US` folder.

Sorry I didn't append the fastlane summary

```
[14:57:07]: Failed verification of all screenshots deleted... 7 screenshot(s) still exist
```

fastlane summary 
----------------------------------------------------------------------------------
| Step | Action                                                                      | Time (in s) |
| ------ | ------------------------------------------------------ | ---------------------- |
| 1-97 | A whole bunch of pre-build prep and tests            | Lots |
| 98   | Switch to deploy_prod_ios lane                                | 0           |
| 99   | [REDACTED_LANE]                                                          | 0           |
| 💥   | deliver                                                                      | 18          |
| 101  | update_build_status                                                 | 0           |
----------------------------------------------------------------------------------

```[14:57:07]: fastlane finished with errors
[!] Failed verification of all screenshots deleted... 7 screenshot(s) still exist
Build step 'Execute shell' marked build as failure
[...]
```

**Comment by ainame** [comment]
@dlakes-hx 

>  The only assets on the prior release are for en-US, and also the same on the pending release in the fastlane/screenshots/en-US folder.

Can you please elaborate on what changes/diffs you are trying in screenshots updates?
For example, 

```
* v 4.0.0 (released) has 7 screenshots for 6.5 inch screen in "en-US"
* v 4.0.1 (you are perparing) is going to replace existing 7 screenshots for 6.5 inch screen in "en-US" with new ones
* etc...
```

I'm a bit confused because your prior description's log shows 35 images not deleted but the summary you posted at last shows 7 remained...  

> [14:16:57]: Failed verification of all screenshots deleted... 35 screenshot(s) still exist
> [!] Failed verification of all screenshots deleted... 7 screenshot(s) still exist

> In any cases, manually cleaning up screenshots on App Store Connect should resolve the issue, for the time being, I believe.

Again, if you remove all the screenshots on App Store Connect manually, it should resolve the issue for the time being. Make sure screenshots are removed by reloading App Store Connect page. It's async-operation so that it might fail with delays sometimes.

**Comment by dlakes-hx** [comment]
@ainame  I'll try and unpack each bit:

> I'm a bit confused because your prior description's log shows 35 images not deleted but the summary you posted at last shows 7 remained...

Yes, you're correct. I pulled that second log entry from a different app build, my apologies for the confusion. We do whitelabel builds, and I just picked the wrong log output. But all our builds are showing the same symptoms.

> Can you please elaborate on what changes/diffs you are trying in screenshots updates?

For the original example

* v 4.0.0 (released) has 35 screenshots spread among several device sizes
* v 4.0.1 (preparing) is going to replace those 35. It currently has no images in the App Store Connect portal.

> Again, if you remove all the screenshots on App Store Connect manually, it should resolve the issue for the time being

I'm unable to delete from `4.0.0`, no option to do so. And there are no images in `4.0.1` to delete.

I could see the argument that we need to be performing that check before passing the `overwrite_screenshots` flag. However, this approach was working for us without issue on `2.152.0` without any problems. 

If it's expected that we need to perform a `deliver init` and check image count beforehand, it would help to have the documentation updated to reflect that as a break from prior behavior.

**Comment by ainame** [comment]
@dlakes-hx 

Thank you for the explanations!

I think this is not expected behaviour. In your cases, once you deleted all the images for v4.0.1 on App Store Connect somehow, there shouldn't be any deletions. As far as I'm aware, you don't need to do this >  we need to perform a deliver init and check image count beforehand

The thing I see strange in your situation is that even you have no images on v4.0.1, `deliver` seemed to try deleting 35 images...? `deliver` should point the current edit version to delete/upload screenshots and it shouldn't be `v4.0.0` that's released already. So if you say there's no image on App Store Connect, `deliver` should go L110 at the first try without having to execute any deletions... 

https://github.com/fastlane/fastlane/blob/5ee9f2e1f1d602e2af129e94eec2313091448abf/deliver/lib/deliver/upload_screenshots.rb#L96-L112

To get better insights, can you please run your submit-command with `--verbose` option which outputs more info for debugging and share the log (around deliver's log) with here if possible...?

**Comment by dlakes-hx** [comment]
@ainame hopefully this provides some insight

<details>
<summary>Verbose error output</summary>
<pre>
INFO [2020-10-06 11:23:00.10]: ---------------------
INFO [2020-10-06 11:23:00.10]: --- Step: deliver ---
INFO [2020-10-06 11:23:00.10]: ---------------------
INFO [2020-10-06 11:23:00.10]: Login to App Store Connect (REDACTED)
Reading keychain entry, because either user or password were empty
Loading session from '/Users/REDACTED/.fastlane/spaceship/REDACTED/cookie'
Looking for App Store Connect Team with ID REDACTED
INFO [2020-10-06 11:23:01.40]: Login successful
+--------------------------------------+----------------------------------------------------------------+
|                                        deliver 2.162.0 Summary                                        |
+--------------------------------------+----------------------------------------------------------------+
| ipa                                  | REDACTED.ipa                                                   |
| primary_category                     | REDACTED                                                       |
| secondary_category                   | REDACTED                                                       |
| overwrite_screenshots                | true                                                           |
| force                                | true                                                           |
| app_identifier                       | REDACTED                                                       |
| screenshots_path                     | ./fastlane/screenshots                                         |
| metadata_path                        | ./fastlane/metadata                                            |
| app_version                          | 4.0.1                                                          |
| username                             | REDACTED                                                       |
| platform                             | ios                                                            |
| edit_live                            | false                                                          |
| use_live_version                     | false                                                          |
| skip_binary_upload                   | false                                                          |
| skip_screenshots                     | false                                                          |
| skip_metadata                        | false                                                          |
| skip_app_version_update              | false                                                          |
| submit_for_review                    | false                                                          |
| reject_if_possible                   | false                                                          |
| automatic_release                    | false                                                          |
| phased_release                       | false                                                          |
| reset_ratings                        | false                                                          |
| run_precheck_before_submit           | true                                                           |
| precheck_default_rule_level          | warn                                                           |
| ignore_language_directory_validation | false                                                          |
| precheck_include_in_app_purchases    | true                                                           |
+--------------------------------------+----------------------------------------------------------------+

INFO [2020-10-06 11:23:03.62]: Making sure the latest version on App Store Connect matches '4.0.1'...
INFO [2020-10-06 11:23:04.02]: '4.0.1' is the latest version on App Store Connect
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/description.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/keywords.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/support_url.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/marketing_url.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/promotional_text.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/en-US/privacy_url.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/copyright.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/review_information/first_name.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/review_information/last_name.txt'...
INFO [2020-10-06 11:23:04.02]: Loading './fastlane/metadata/review_information/phone_number.txt'...
WARN [2020-10-06 11:23:06.14]: Will begin uploading metadata for '4.0.1' on App Store Connect
INFO [2020-10-06 11:23:06.14]: Uploading metadata to App Store Connect for version
INFO [2020-10-06 11:23:11.46]: Uploading app review information to App Store Connect
WARN [2020-10-06 11:23:12.80]: Will begin uploading snapshots for '4.0.1' on App Store Connect
INFO [2020-10-06 11:23:12.80]: Starting with the upload of screenshots...
WARN [2020-10-06 11:23:14.06]: Number of screenshots not deleted: 7
ERROR [2020-10-06 11:23:14.06]: Failed to delete all screenshots... Tries remaining: 4
WARN [2020-10-06 11:23:14.78]: Number of screenshots not deleted: 7
ERROR [2020-10-06 11:23:14.78]: Failed to delete all screenshots... Tries remaining: 3
WARN [2020-10-06 11:23:15.48]: Number of screenshots not deleted: 7
ERROR [2020-10-06 11:23:15.48]: Failed to delete all screenshots... Tries remaining: 2
WARN [2020-10-06 11:23:16.22]: Number of screenshots not deleted: 7
ERROR [2020-10-06 11:23:16.22]: Failed to delete all screenshots... Tries remaining: 1
WARN [2020-10-06 11:23:17.02]: Number of screenshots not deleted: 7
WARN [2020-10-06 11:23:17.02]: Reporting build status error for lane: deploy_prod_ios
INFO [2020-10-06 11:23:17.02]: Backtrace: ["/Users/redacted/jenkins/workspace/REDACTED/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:104:in `delete_screenshots'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:107:in `delete_screenshots'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/upload_screenshots.rb:36:in `upload'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/runner.rb:146:in `upload_metadata'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/deliver/lib/deliver/runner.rb:54:in `run'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:17:in `run'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'", "Fastfile:446:in `block in parsing_binding'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane.rb:33:in `call'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'", "Fastfile:722:in `block in parsing_binding'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane.rb:33:in `call'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/runner.rb:45:in `execute'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/bin/fastlane:23:in `<top (required)>'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<main>'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `eval'", "/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/bin/ruby_executable_hooks:24:in `<main>'"]
INFO [2020-10-06 11:23:17.02]: No error info available. Returning the exception as message
ERROR [2020-10-06 11:23:17.02]: Failed verification of all screenshots deleted... 7 screenshot(s) still exist
INFO [2020-10-06 11:23:17.02]: ---------------------------------
INFO [2020-10-06 11:23:17.02]: --- Step: update_build_status ---
INFO [2020-10-06 11:23:17.02]: ---------------------------------
INFO [2020-10-06 11:23:17.02]: Updating build status record
INFO [2020-10-06 11:23:17.22]: Successfully updated build status record!
WARN [2020-10-06 11:23:17.22]: Lane Context:
INFO [2020-10-06 11:23:17.22]: REDACTED
ERROR [2020-10-06 11:23:17.22]: Failed verification of all screenshots deleted... 7 screenshot(s) still exist
INFO [2020-10-06 11:23:17.22]: Successfully generated documentation at path '/Users/redacted/jenkins/workspace/Redacted/fastlane/README.md'

+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1-99 | Redacted setup steps                                          | 0           |
| 💥   | deliver                                                       | 16          |
| 101  | update_build_status                                           | 0           |
+------+---------------------------------------------------------------+-------------+

DEBUG [2020-10-06 11:23:17.23]: All plugins are up to date
ERROR [2020-10-06 11:23:17.23]: fastlane finished with errors
DEBUG [2020-10-06 11:23:17.23]: All plugins are up to date
/Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': [!] Failed verification of all screenshots deleted... 7 screenshot(s) still exist (FastlaneCore::Interface::FastlaneError)
	from /Users/redacted/jenkins/workspace/Redacted/vendor/bundle/ruby/2.6.0/gems/fastlane-2.162.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
</pre></details>

Two points of interest:

* I double checked a prior log for this app, and it was showing the

    ```
    Failed verification of all screenshots deleted... 35 screenshot(s) still exist
    ```
    Note the new count is 7. 
* We're using app store connect team ids for most of our builds, but are not yet using the new App Store Connect API token. 
  * I was able to verify that the team id and the user credentials match that expected for the app being build

**Comment by ainame** [comment]
@dlakes-hx 

Assuming you have set `--verbose` option properly to run your submit-command, what I can see from the logs is that `deliver` doesn't even try to delete screenshots. It should have contained logs like this if it's trying to delete some.

```
INFO [2020-10-06 17:01:33.48]: Starting with the upload of screenshots...
DEBUG [2020-10-06 17:01:37.15]: Queued delete sceeenshot job for en-US APP_IPHONE_65 2a845303-b164-437c-b940-0157c0c42201
DEBUG [2020-10-06 17:01:37.15]: Queued delete sceeenshot job for en-US APP_IPHONE_65 f9a44a6d-1d8f-4d98-91dd-cd6ccbb47e80
DEBUG [2020-10-06 17:01:37.15]: Queued delete sceeenshot job for en-US APP_IPHONE_65 146388d7-7b38-4cf3-afb1-c7d0cf0a162b
DEBUG [2020-10-06 17:01:37.15]: Deleting 'en-US APP_IPHONE_65 2a845303-b164-437c-b940-0157c0c42201'
DEBUG [2020-10-06 17:01:37.15]: Deleting 'en-US APP_IPHONE_65 f9a44a6d-1d8f-4d98-91dd-cd6ccbb47e80'
DEBUG [2020-10-06 17:01:37.16]: Deleting 'en-US APP_IPHONE_65 146388d7-7b38-4cf3-afb1-c7d0cf0a162b'
```

https://github.com/fastlane/fastlane/blob/5ee9f2e1f1d602e2af129e94eec2313091448abf/deliver/lib/deliver/upload_screenshots.rb#L76

As I told in the previous message, there is a tricky behaviour that `deliver` skips to delete images on particular localisations where local data's directory structure don't match.

https://github.com/fastlane/fastlane/blob/5ee9f2e1f1d602e2af129e94eec2313091448abf/deliver/lib/deliver/upload_screenshots.rb#L87-L88

So can you please make sure if there's no localization other than en-US? 😅 This is the last thing I can help you to resolve the issue... Otherwise, I think it is likely to be an issue related to Connect API on either fastlane's API client or Apple's end.

<img width="327" alt="Screenshot 2020-10-06 at 17 20 50" src="https://user-images.githubusercontent.com/748949/95229600-abc48900-07f8-11eb-9822-84732e924b1b.png">

**Comment by dlakes-hx** [comment]
@ainame It looks like this may be a matter of "fixed the bug". 

Our builds have a shared workspace, so I think when you asked about the directory structure last time I must have been seeing structure from a subsequent build, and not realized it.

I bumped up logging on the call, and it appears for that particular build, the directory is:
`fastlane/screenshots`. *However* it has no contents, including the localization `en-US` folder.

So odds are the overwrites/deletes were failing before, the system was just swallowing it. We'll update our process to only toggle `overwrite_screenshots` when there's something in the directory.

I'm personally happy to close this one out. Thanks so much for your help sorting the minor functionality change and my boneheaded overlook @ainame !

**Comment by ainame** [comment]
Haha, It's my pleasure🙂

**Comment by ianbytchek** [comment]
Thanks for reporting this and investigating @dlakes-hx! 

Hey @ainame, this is apparently the same thing I mentioned in #16972. I changed the default locale from en-GB to en-US awhile ago and looks like that's precisely what causes the issue as you guys have established. I think one bit that can be improved if `screenshots` could support `default` locale like metadata does. My metadata looks like follows and when I upload it fastlane reports the following:

<img width="264" alt="image" src="https://user-images.githubusercontent.com/1086845/95577376-149f4180-0a43-11eb-95a6-42d3714fc09d.png">

```sh
...
INFO [2020-10-09 15:16:47.68]: Detected languages: ["default", "en-US"]
WARN [2020-10-09 15:16:53.94]: Will begin uploading metadata for '2.2.4' on App Store Connect
...
```

This is very neat, effectively I guess I should get rid of en-US symlink and not rely on manually specifying the locale. However, trying the same approach with screenshots I get something like `locale "default" is not supported`.

So, can screenshots support `default` locale? What do you think?

**Comment by Pinto88** [comment]
Thanks for the information keep suggesting such informative post [my prepaid center](url=https://www.myprepaidcenter.us/)

**Comment by ainame** [comment]
@ianbytchek 

Glad to hear that you figured out your issue. Hope you are okay with it now🤞 
Regarding your suggestion, it would be definitely possible to implement, I guess. 

Since I'm not a core contributor at all, I can decide nothing for new functionalities. You should create a feature request ticket. If the spec is fixed and people agree with it, I can pick it up to implement.

Unlike metadata's `default` support, the union of both default and a language's screenshots are likely to be undesirable. So the only thing it benefits is just that you can set a directory name to `default`. However, the downside of it that I can think of is that it requires validation for directory names to avoid edge cases and it might be confused for even developers and its locale can't be recognised unless you preview the image. 

For example, if the screenshots directory has duplicated directory named same as the `default` locale. Let's say there are `default` and `en-US` directories and the default locale is `en-US`. `deliver` should fail in that case.  I think that kind of discussion should happen before adding a new behaviour😄

**Comment by spherefoundry** [comment]
Hey @ainame, 

I'm facing a variation of the same problem. By looking at the code I think the verification part in `delete_screenshots` has a small logic flaw. It counts screenshots for all locales, even those it didn't even try to delete. 

So IMO the behaviour when the `overwrite_screenshots` flag is set should be on of the two:

1) Remove all screenshots for all locales present in AppStore Connect.
2) Completely ignore screenshots for additional locales that are present in AppStore Connect, and just sync the screenshots for the locales provided.

What do you think?

**Comment by ainame** [comment]
Yes, I've recognised this issue. In my opinion, I'd prefer to keep the existing behaviour for existing fastlane users so that PR can get merged without deep discussions. So I would go with 2 (if I understand your description correct). I might implement it as it's been a bit annoying me while testing `deliver`🙂 

If you insist 1 is the go-to option, would you be able to convince Fastlane core contributors (I am not) and users...? I would recommend you open a discussion here. I can join it.
https://github.com/fastlane/fastlane/discussions

**Comment by ainame** [comment]
@spherefoundry 

Yes, I've recognised this issue. In my opinion, I'd prefer to keep the existing behaviour for existing fastlane users so that PR can get merged without deep discussions. So I would go with 2 (if I understand your description correct). I might implement it as it's been a bit annoying me while testing `deliver`🙂 

If you insist 1 is the go-to option, would you be able to convince Fastlane core contributors (I am not) and users...? I would recommend you open a discussion here. I can join it.
https://github.com/fastlane/fastlane/discussions

**Comment by spherefoundry** [comment]
@ainame 

I think both options work for me 😄  If option 2 is easier because of pre-existing behaviour, then I would say we have a winner. 

Something that would additionally help us a lot would be to get some more logs around the filtering part, as to understand better which screenshots ware not delated.

BTW: I'm really sorry for not being able to help you our here with a PR of my own. The Contributors Agreement is standing in my way 😞 It shouldn't be a problem to get it signed, but nothing is easy when legal is involved.

**Comment by JimHowell20** [comment]
I keep getting this error on fastlane version 2.164.0.

**Comment by squallstar** [comment]
I am getting the same error on 2.166.
